src/app/rpmostree-compose-builtin-tree.cxx

```diff
@@ -297,6 +297,13 @@ set_repos_dir (DnfContext *dnfctx, rpmostreecxx::Treefile &treefile, int workdir
   if (!repovars.empty ())
     g_ptr_array_add (var_dirs, (char *)repovars.c_str ());
 
+  g_autofree char *source_vars = NULL;
+  if (opt_source_root)
+    {
+      source_vars = g_build_filename (opt_source_root, "etc/dnf/vars", NULL);
+      g_ptr_array_add (var_dirs, (char *)source_vars);
```

Comment:
What are the semantics if both the treefile and `/etc/dnf/vars` define the same variable? If I'm reading https://github.com/rpm-software-management/libdnf/blob/998efa8f41188f9dc4b08b74086c2c3e2a38f73a/libdnf/conf/ConfigMain.cpp#L662 right, I think the latter one wins, so in this case it'd be `/etc/dnf/vars`, but I think we want the treefile to win, right? So that would argue for moving this up.

Reply:
Fair, though I think when one is using `--source-root` then we should discourage setting things in the treefile. Of course for some of our projects there may be a transition phase, but...I think we could also just error out actually.